cylc/rose/utilities.py

```diff
@@ -325,6 +330,103 @@ def merge_rose_cylc_suite_install_conf(old, new):
     return old
 
 
+def invalid_defines_check(defines: List) -> None:
+    """Check for defines which do not contain an = and therefore cannot be
+    valid
+
+    Examples:
+
+        # A single invalid define:
+        >>> import pytest
+        >>> with pytest.raises(InvalidDefineError, match=r'\\* foo'):
+        ...    invalid_defines_check(['foo'])
+
+        # Two invalid defines and one valid one:
+        >>> with pytest.raises(
+        ...     InvalidDefineError, match=r'\\* foo.*\\n.* \\* bar'
+        ... ):
+        ...     invalid_defines_check(['foo', 'bar52', 'baz=442'])
+
+        # No invalid defines
+        >>> invalid_defines_check(['foo=12'])
+    """
+    invalid_defines = []
+    for define in defines:
+        if parse_cli_defines(define) == 'Invalid':
+            invalid_defines.append(define)
+    if invalid_defines:
+        msg = 'Invalid Suite Defines (should contain an =)'
+        for define in invalid_defines:
+            msg += f'\n * {define}'
+        raise InvalidDefineError(msg)
+
+
+def parse_cli_defines(define: str) -> Union[
+    bool, str, Tuple[
+        List[Union[str, Any]],
+        Union[str, Any],
+        Union[str, Any],
+    ]
+]:
+    """Parse a define string.
+
+    Args:
+        define:
+            A string in one of two forms:
+            - `key = "value"`
+            - `[section]key = "value"`
+
+            With optional `!` and `!!` prepended, indicating an ignored state,
+            which should lead to a warning being logged.
+
+    Returns:
+        False: If state is ignored or trigger-ignored, otherwise...
+        (keys, value, state)
+
+    Examples:
+        # Top level key
+        >>> parse_cli_defines('root-dir = "foo"')
+        (['root-dir'], '"foo"', '')
+
+        # Marked as ignored
+        >>> parse_cli_defines('!root-dir = "foo"')
+        False
+
+        # Inside a section
+        >>> parse_cli_defines('[section]orange = "segment"')
+        (['section', 'orange'], '"segment"', '')
+    """
+    match = re.match(
+        (
+            r'^\[(?P<section>.*)\](?P<state>!{0,2})'
+            r'(?P<key>.*)\s*=\s*(?P<value>.*)'
+        ),
+        define
+    )
+    if match:
+        groupdict = match.groupdict()
+        keys = [groupdict['section'].strip(), groupdict['key'].strip()]
+    else:
+        # Doesn't have a section:
+        match = re.match(
+            r'^(?P<state>!{0,2})(?P<key>.*)\s*=\s*(?P<value>.*)', define)
+        if match:
+            groupdict = match.groupdict()
+            keys = [groupdict['key'].strip()]
+        else:
+            # This seems like it ought to be an error,
+            # But behaviour is consistent with Rose 2019
+            # See:
+            return "Invalid"
+
+    if match['state'] in ['!', '!!']:
+        LOG.warning(
+            'CLI opts set to ignored or trigger-ignored will be ignored.')
```

Comment:
Is that a thing? (Use the ignore syntax on the command line?) Why would anyone do that?  And if they do do it, should it need a warning?

Reply:
Yes, this is a "thing", in Rose ignored items still exist in the configuration, just with a different state so could still be functional in some strange way.

I don't think we need a warning here, you'd have to go out of your way to add an ignored config, if you do we can only assume you know what you're doing.